From 4341fb875cf8326faedd000ec8170c38e7a2895a Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Thu, 13 Nov 2014 04:34:37 -0500
Subject: [PATCH 068/118] cat, diff: avoid double slashes in paths
 (RHBZ#1151910).

In full_path, skip the trailing slash in the base directory when
different than "/", as the slash will eventually be added when building
the resulting path.
(cherry picked from commit 87941d183f4a96b4d936c2e94a79a60288f0dc74)
---
 cat/virt-ls.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/cat/virt-ls.c b/cat/virt-ls.c
index e6dea23..871b545 100644
--- a/cat/virt-ls.c
+++ b/cat/virt-ls.c
@@ -538,13 +538,18 @@ full_path (const char *dir, const char *name)
 {
   int r;
   char *path;
+  int len;
+
+  len = strlen (dir);
+  if (len > 0 && dir[len - 1] == '/')
+    --len;
 
   if (STREQ (dir, "/"))
     r = asprintf (&path, "/%s", name ? name : "");
   else if (name)
-    r = asprintf (&path, "%s/%s", dir, name);
+    r = asprintf (&path, "%.*s/%s", len, dir, name);
   else
-    r = asprintf (&path, "%s", dir);
+    r = asprintf (&path, "%.*s", len, dir);
 
   if (r == -1) {
     perror ("asprintf");
-- 
2.7.4

